test/unit/data.js

```diff
@@ -874,3 +874,18 @@ test( "Check that the expando is removed when there's no more data", function()
 		}
 	}
 });
+
+test( ".data(prop) does not create expando", function() {
+	expect( 2 );
+
+	var div = jQuery( "<div/>" );
+
+	div.data("foo");
+	equal( false, jQuery.hasData( div[0] ) );
+	// Make sure no expando has been added
+	for ( key in div[ 0 ] ) {
```

Comment:
> I really don't like this form. :)

This is why we have a code-style, i don't like a lot of things in current code base, as i'm sure we all do, but we choose to follow it.

> (i.e. this should be done exactly as this PR does it)

No, test should trigger the error with minimum overhead, any other requirements are subjective